src/textual/filter.py

```diff
@@ -188,9 +190,18 @@ def __init__(self, terminal_theme: TerminalTheme):
         Args:
             terminal_theme: A rich terminal theme.
         """
-        self.terminal_theme = terminal_theme
+        self._terminal_theme = terminal_theme
+        self._truecolor_cache: LRUCache[Style, Style] = LRUCache(maxsize=1024)
+
+    @property
+    def theme(self) -> TerminalTheme:
+        return self._terminal_theme
+
+    @theme.setter
+    def theme(self, theme: TerminalTheme) -> None:
```

Comment:
Rather than mutate this object, can we reconstruct the `ANSIToTruecolor` when the theme changes?

That way we can keep the `lru_cache`, which is faster. Filters run on ever segment, so the extra performance is worthwhile.

Reply:
I've made this change but for some reason CI won't kick off. I think GitHub is having trouble - will try again tomorrow.